### PR TITLE
Arm/feature/tx index sync

### PR DESCRIPTION
**Pull request by arminsabouri**
Please review carefully!

The main changes are:
* new bitcoin client factory 
* new btc server endpoint to sync node to new checkpoints
* calling said enpoint when async bitcoin worker recives new tip
* Fixes regarding bitcoin conf depth

General note as I was writing this code: 
Pegins should be valid if they are confirmed at a height greater than the pegin conf depth. For example if conf_depth = 1. Pegins are valid if they are confirmed in a block at height = (tip - conf_depth - 1). Meaning you need two confirmations.
in other words our checkpoint block should be =(tip - conf_depth - 1). Not tip - conf_depth.

A small visual: 

[Pegins           
are valid
here]                           
Checkpoint Block -> Tip - 1 -> Tip

### CODE REVIEW

**Comment by scottmillner** [comment]
> Please review carefully!
> 
> The main changes are:
> 
> * new bitcoin client factory
> * new btc server endpoint to sync node to new checkpoints
> * calling said enpoint when async bitcoin worker recives new tip
> * Fixes regarding bitcoin conf depth
> 
> General note as I was writing this code: Pegins should be valid if they are confirmed at a height greater than the pegin conf depth. For example if conf_depth = 1. Pegins are valid if they are confirmed in a block at height = (tip - conf_depth - 1). Meaning you need two confirmations. in other words our checkpoint block should be =(tip - conf_depth - 1). Not tip - conf_depth.
> 
> A small visual:
> 
> [Pegins are valid here] Checkpoint Block -> Tip - 1 -> Tip

Per our convo earlier... pegins are valid at `tip - conf_depth` right?

**Review by rwlockbg — DISMISSED** [review]

**Comment by arminsabouri** [comment]
> > Please review carefully!
> > The main changes are:
> > 
> > * new bitcoin client factory
> > * new btc server endpoint to sync node to new checkpoints
> > * calling said enpoint when async bitcoin worker recives new tip
> > * Fixes regarding bitcoin conf depth
> > 
> > General note as I was writing this code: Pegins should be valid if they are confirmed at a height greater than the pegin conf depth. For example if conf_depth = 1. Pegins are valid if they are confirmed in a block at height = (tip - conf_depth - 1). Meaning you need two confirmations. in other words our checkpoint block should be =(tip - conf_depth - 1). Not tip - conf_depth.
> > A small visual:
> > [Pegins are valid here] Checkpoint Block -> Tip - 1 -> Tip
> 
> Per our convo earlier... pegins are valid at `tip - conf_depth` right?

Correct! I will keep this in draft till I fix tis

**Review by scottmillner — DISMISSED** [review]

**Review by scottmillner — APPROVED** [review]

**Comment by arminsabouri** [comment]
Testing results are all green on my end
including functional test + pegin/pegout
